tests/Benchmark/CreateSignatureBench.php

```diff
@@ -22,6 +22,6 @@ public function initialize(array $params): void
 
     protected function runBenchmark(): void
     {
-        $this->algorithm->sign(self::PAYLOAD, $this->key);
+        $void = $this->algorithm->sign(self::PAYLOAD, $this->key);
```

Comment:
Unfortunately the `(void)` cast doesn't compile on PHP < 8.5

Reply:
One last question here: why does our static analysis not pick this up?

Reply:
I'm taking a guess here: because `Signer` is not marked `@immutable`.
Shall we make it so?

Reply:
Deterministic output is something for `@pure`, not `@immutable`, isn't it?

Reply:
I am sorry but what you say differs from what I read online:

https://psalm.dev/articles/immutability-and-beyond

> `@immutable` guarantees that all class properties can be written to just once in the constructor, and all instance methods won’t mutate state outside the class

Depending on a mutable state (like randomness) doesn't mean that the class itself mutate such state

PHPStan has an even more loose definition https://phpstan.org/writing-php-code/phpdocs-basics#immutable-classes

> `@immutable` or `@readonly` on the class can be used to make PHPStan treat every property of that class as being readonly.